src/backend/joanie/core/serializers/client.py

```diff
@@ -613,9 +613,35 @@ def create(self, validated_data, **kwargs):
             message = f'A course run with id "{course_run_id}" does not exist.'
             raise serializers.ValidationError({"__all__": [message]}) from exception
 
+        # The related course run must be opened to create the enrollment or reactivate it.
+        if course_run.state["priority"] > models.CourseState.ARCHIVED_OPEN:
+            message = _(
+                "You are not allowed to enroll to a course run not opened for enrollment."
+            )
+            raise serializers.ValidationError({"__all__": [message]})
+
         validated_data["course_run"] = course_run
 
-        return super().create(validated_data=validated_data)
+        instance = super().create(validated_data=validated_data)
+
+        # The user should not be enrolled in another opened course run of the same course.
+        if (
+            instance.is_active
+            and instance.user.enrollments.exclude(id=instance.id)
+            .filter(
+                course_run__course=instance.course_run.course,
+                course_run__end__gte=timezone.now(),
+                is_active=True,
+            )
+            .exists()
+        ):
+            message = _(
+                "You are already enrolled to an opened course run "
+                f'for the course "{instance.course_run.course.title}".'
+            )
+            raise serializers.ValidationError({"user": [message]})
```

Comment:
We should check that before the creation of the enrollment.

Reply:
I will update it ;) thanks